gosip panic

the stack is below:
```go

panic: invalid argument to Intn

goroutine 29 [running]:
math/rand.(*Rand).Intn(0xc0000f8180, 0x0)
	/usr/lib/go/src/math/rand/rand.go:180 +0x72
math/rand.Intn(0x0)
	/usr/lib/go/src/math/rand/rand.go:453 +0x34
github.com/ghettovoice/gosip/transport.(*layer).Send(0xc00041a640, {0x1a3ae58, 0xc00043a420})
	/home/weihg/code/go/pkg/mod/github.com/ghettovoice/gosip@v0.0.0-20231212083318-2c9f9eaee8d5/transport/layer.go:239 +0x4a5
github.com/cloudwebrtc/go-sip-ua/pkg/stack.(*SipStack).Send(0xc000000540, {0x1a3ae58, 0xc00043a420})
	/home/weihg/code/go/pkg/mod/github.com/yangjuncode/go-sip-ua@v1.1.8/pkg/stack/stack.go:435 +0x22e
github.com/cloudwebrtc/go-sip-ua/pkg/stack.(*sipTransport).Send(0xc000426468, {0x1a3ae58, 0xc00043a420})
	/home/weihg/code/go/pkg/mod/github.com/yangjuncode/go-sip-ua@v1.1.8/pkg/stack/stack.go:572 +0x45
github.com/ghettovoice/gosip/transaction.(*clientTx).Init(0xc00031a000)
	/home/weihg/code/go/pkg/mod/github.com/ghettovoice/gosip@v0.0.0-20231212083318-2c9f9eaee8d5/transaction/client_tx.go:92 +0xc3
github.com/ghettovoice/gosip/transaction.(*layer).Request(0xc00051ca00, {0x1a3b480, 0xc00043a420})
	/home/weihg/code/go/pkg/mod/github.com/ghettovoice/gosip@v0.0.0-20231212083318-2c9f9eaee8d5/transaction/layer.go:140 +0x4fc

```

--- a/transport/layer.go
+++ b/transport/layer.go
@@ -235,7 +235,7 @@ func (tpl *layer) Send(msg sip.Message) error {
 
 		// rewrite sent-by port
 		if viaHop.Port == nil {
-			if ports, ok := tpl.listenPorts[network]; ok {
+			if ports, ok := tpl.listenPorts[network]; ok && (len(ports) > 0) {
 				port := ports[rand.Intn(len(ports))]
 				viaHop.Port = &port
 			} else {

